src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -457,11 +461,22 @@ public sealed override IEnumerable<CombinedDependencyListEntry> GetConditionalSt
                         if (!isStaticInterfaceMethod && !needsDependenciesForInstanceInterfaceMethodImpls)
                             continue;
 
+                        MethodDesc interfaceMethodDefinition = interfaceMethod;
+                        if (interfaceType != interfaceDefinitionType)
+                            interfaceMethodDefinition = factory.TypeSystemContext.GetMethodForInstantiatedType(interfaceMethodDefinition.GetTypicalMethodDefinition(), (InstantiatedType)interfaceDefinitionType);
```

Comment:
Not sure what `GetTypicalMethodDefinition` does. I think we get here when we have something like `I1<T>` and `I1<string>`, where `interfaceType = I1<string>`. And if `I1<T>` looks like

```
interface I1<T>
{
    void M(T t);
}
```

then `interfaceMethod = I1<string>.M(string)`?

Then does this statement get `I1<T>.M(T t)`?

Reply:
It gets the method definition on the uninstantiated version of the owning type:

`I1<!0>.Method`.GetTypicalMethodDefinition == `I1<T>.Method`
`I1<object>.Method`.GetTypicalMethodDefinition == `I1<T>.Method`
`I1<List<!0>>.Method`.GetTypicalMethodDefinition == `I1<T>.Method`

And also (but it might not be important here):

`I1<!0>.Method<!1>`.GetTypicalMethodDefinition == `I1<T>.Method<U>`
`I1<object>.Method<!1>`.GetTypicalMethodDefinition == `I1<T>.Method<U>`
`I1<List<!0>>.Method<object>`.GetTypicalMethodDefinition == `I1<T>.Method<U>`